domain/model/service/modelservice.go

```diff
@@ -157,3 +212,23 @@ func (s *ModelService) GetStatus(ctx context.Context) (model.StatusInfo, error)
 		Since:  now,
 	}, nil
 }
+
+// GetEnvironProviderVersion retrieves the version of the environment provider associated with the model.
+//
+// Errors:
+// - [modelerrors.NotFound]: Returned if the model does not exist.
+func (s *ModelService) GetEnvironProviderVersion(ctx context.Context) (int, error) {
+	model, err := s.modelSt.GetModel(ctx)
```

Comment:
I think we have defined the new state func for getting the cloud type but we're not using it yet?

Reply:
yes thanks for spotting this tom :)